src/scr.c

```diff
@@ -1708,22 +1713,39 @@ int SCR_Route_file(const char* file, char* newfile)
 
     /* build absolute path to file */
     scr_path* path_abs = scr_path_from_str(file);
-    if (! scr_path_is_absolute(path_abs)) {
-      /* the path is not absolute, so prepend the current working directory */
-      char cwd[SCR_MAX_FILENAME];
-      if (scr_getcwd(cwd, sizeof(cwd)) == SCR_SUCCESS) {
-        scr_path_prepend_str(path_abs, cwd);
-      } else {
-        /* problem acquiring current working directory */
-        scr_abort(-1, "Failed to build absolute path to %s @ %s:%d",
-          file, __FILE__, __LINE__
-        );
+    if (scr_preserve_directories) {
+      if (! scr_path_is_absolute(path_abs)) {
+        /* the path is not absolute, so prepend the current working directory */
+        char cwd[SCR_MAX_FILENAME];
+        if (scr_getcwd(cwd, sizeof(cwd)) == SCR_SUCCESS) {
```

Comment:
Interesting point.  This code was already in there, and using prefix as the base dir didn't occur to me.  For now, I think I'll keep with cwd, though.  Since existing apps are set up to expect their files to show up in path relative to the current working dir, this option avoids making them change that.  The one requirement is that whatever path they end up with must be under prefix somewhere.